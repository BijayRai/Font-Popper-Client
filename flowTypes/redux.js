// @flow
import type { Dispatch } from 'redux'
import type { User } from './User'

export type Action = {
  type: string,
  payload?: Promise<any> | any,
  error?: boolean,
  user?: any
}

export type ThunkAction = {
  dispatch: Dispatch<Action>,
  getState: () => any,
  extraArgument: (any) => any;
}

export type DispatchAction = Dispatch<Action>

export type ReduxForm = {
  anyTouched: boolean,
  array: any,
  asyncValidate: Function,
  asyncValidating: boolean,
  autofill: Function,
  blur: Function,
  change: Function,
  clearAsyncError: Function,
  clearSubmit: Function,
  clearSubmitErrors: Function,
  destroy: Function,
  error: any,
  errorMessage: string,
  form: string,
  handleSubmit: Function,
  initialValues: any,
  initialized: boolean,
  invalid: boolean,
  pristine: boolean,
  pure: boolean,
  reset: Function,
  submit: Function,
  submitFailed: boolean,
  submitSucceeded: boolean,
  submitting: boolean,
  touch: Function,
  triggerSubmit: any,
  untouch: Function,
  valid: boolean,
  validate: Function,
  warning: any
}

export type ReduxFormField = {
  value: string,
  active: boolean,
  autofill: Function,
  autofilled: ?boolean,
  checked: ?boolean,
  dirty: boolean,
  error: string,
  initialValue: any,
  invalid: boolean,
  name: string,
  onBlur(valueOrEvent: any): Function,
  onChange(valueOrEvent: any): Function,
  onDragStart: Function,
  onDrop: Function,
  onFocus: Function,
  onUpdate: Function,
  pristine: boolean,
  touched: boolean,
  valid: boolean,
  visited: boolean
}

export type InputField = {
  name: string,
  onBlur: Function,
  onChange: Function,
  onDragStart: Function,
  onDrop: Function,
  onFocus: Function,
  value: any
}
export type InputMeta = {
  active: boolean,
  asyncValidating: boolean,
  autofilled: boolean,
  dirty: boolean,
  dispatch: Function,
  error: any,
  form: string,
  initial: any,
  invalid: boolean,
  pristine: boolean,
  submitFailed: boolean,
  submitting: boolean,
  touched: boolean,
  valid: boolean,
  warning: any
}
export type RenderField = {
  input: InputField,
  label: string,
  meta: InputMeta,
  type: string,
  className?: string,
  value?: any,
  id?: string,
  name?: string,
  key?: string
}

export type ReduxStore = {
  dispatch: Dispatch,
  getState: () => any,
  liftedStore: any,
  replaceReducer: (any) => any,
  subscribe: (any) => any,
  'Symbol(observable)': Function
}

type pageContent = {
  ids: string[],
  fetching: boolean
}
type pageNumber = {
  [number]: pageContent
}
type Results = {
  [string]: pageNumber
}

type Pages = {
  pages: Results
}

export type State = {
  time: number,
  isSaving: boolean,
  user: User,
  userAccount: any,
  form: any,
  toaster: any,
  pagination: Pages
}
