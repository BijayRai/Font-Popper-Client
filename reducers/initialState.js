export default {
  time: 0,
  isSaving: false,
  user: {
    isAuthenticated: false
  },
  userAccount: {},
  filtered: {
    tags: [],
    stores: []
  }
}
