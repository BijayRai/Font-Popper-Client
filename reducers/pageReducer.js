import { createPaginator } from '../actions/paginator'
// Data shape - RECIEVE STORE PAGE
// pagination: {
//     pages: {
//       1: {
//         ids: [ 'todo1', 'todo2' ],
//         fetching: false
//       },
//     }
// }

export const pagination = createPaginator('/stores', 'stores')
