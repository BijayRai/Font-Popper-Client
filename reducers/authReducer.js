// @flow

import { actionTypes } from '../actions/actionTypes'
import initialState from './initialState'
import type { Action } from '../flowTypes/redux'
import type { User } from '../flowTypes/User'

export const authReducer = (state: User = initialState.user, action: Action): User => {
  switch (action.type) {
    // case actionTypes.SAVE_USER:
    //   return Object.assign({}, state, {
    //     ...action.user,
    //     isAuthenticated: true
    //   })
    case actionTypes.REFRESH_TOKEN:
      return Object.assign({}, state, {
        ...action.user,
        isAuthenticated: true
      })
    case actionTypes.LOG_OUT:
      return {
        isAuthenticated: false
      }
    case actionTypes.USER_LOG_IN_SUCCESS:
      console.log('saved user in redux')

      return Object.assign({}, state, {
        ...action.user,
        isAuthenticated: true
      })
    default:
      return state
  }
}
