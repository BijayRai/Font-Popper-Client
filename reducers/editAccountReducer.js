// TODO add flow when we add user account info
// TODO: Add isSaving functionality
import actionTypes from '../actions/actionTypes'
import initialState from './initialState'

export const editAccountReducer = (state = initialState.userAccount,
                                   action) => {
  switch (action.type) {
    case actionTypes.LOAD_USER_DATA:
      return Object.assign({}, state, {
        data: action.user
      })
    default:
      return state
  }
}
