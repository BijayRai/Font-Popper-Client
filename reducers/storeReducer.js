import { actionTypes } from '../actions/actionTypes'
import initialState from './initialState'

export const storeReducer = (state = initialState.stores, action) => {
  switch (action.type) {
    case actionTypes.LOAD_STORES_SUCCESS:
      // console.log('Store Reducer', action)
      // action from normalizer always has results array on it
      return {
        ...state,
        ...action.data.entities.stores
      }
    default:
      return state
  }
}
