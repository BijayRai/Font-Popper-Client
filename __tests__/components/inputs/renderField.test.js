/* eslint-disable no-unused-vars */
/* eslint-env jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { authReducer } from '../../../reducers/authReducer'
import { createStore, compose, combineReducers, applyMiddleware, bindActionCreators } from 'redux'
import RenderField from '../../../components/inputs/renderField'

const setup = () => {
  // className='form-control inputPwConfirm'
  // name='passwordConfirm'
  // type='password'
  // component={renderField}
  // label='Confirm Password:'
  // const {
  //   input,
  //   label,
  //   type,
  //   meta: {pristine, touched, error, invalid}
  // } = props
  const input = {
    name: 'Name',
    onBlur: () => {}
  }
  const props = {
    className: 'form-control',
    input: input,
    label: 'Name',
    type: 'text',
    meta: { pristine: true, touched: false, error: '', invalid: false }
  }
  return shallow(<RenderField {...props} />)
}
describe('Nav login Block', () => {
  let wrapper = ''

  beforeEach(() => {
    wrapper = setup()
  })

  it('should render input snapshot', () => {
    const input = {
      name: 'name',
      onBlur: () => {}
    }
    const props = {
      className: 'form-control',
      input: input,
      label: 'Name',
      type: 'text',
      meta: { pristine: true, touched: false, error: '', invalid: false }
    }
    const component = renderer.create(
      <RenderField {...props} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correct input type', () => {
    const input = wrapper.find('.form-control[type="text"]')
    expect(input.length).toEqual(1)
  })

  it('should render correct name', () => {
    const input = wrapper.find('.form-control[name="Name"]')
    expect(input.length).toEqual(1)
  })

  it('should render correct label', () => {
    const label = wrapper.find('.control-label')
    expect(label.text()).toEqual('Name')
  })

  it('should render correct parent with classes on !Pristine', () => {
    const inputProp = {
      name: 'Name',
      onBlur: () => {}
    }
    const props = {
      className: 'form-control',
      input: inputProp,
      label: 'Name',
      type: 'text',
      meta: { pristine: false, touched: false, error: '', invalid: false }
    }
    const wrapper = shallow(<RenderField {...props} />)
    const input = wrapper.first()

    expect(input.props().className).toEqual('form-group has-success has-feedback')
  })

  it('should render glyphicon with invalid data and !pristine', () => {
    const inputProp = {
      name: 'Name',
      onBlur: () => {}
    }
    const props = {
      className: 'form-control',
      input: inputProp,
      label: 'Name',
      type: 'text',
      meta: { pristine: false, touched: false, error: '', invalid: true }
    }
    const wrapper = shallow(<RenderField {...props} />)
    const icon = wrapper.find('.glyphicon')

    expect(icon.length).toEqual(1)
  })
})
