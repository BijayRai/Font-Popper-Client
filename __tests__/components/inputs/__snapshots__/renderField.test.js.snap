// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav login Block should render input snapshot 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
  >
    Name
  </label>
  <input
    className="form-control"
    name="name"
    onBlur={[Function]}
    type="text"
  />
</div>
`;
